### PR TITLE
diffusers 0.30.3

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-5909](https://anaconda.atlassian.net/browse/PKG-5909) 
- [Upstream repository](https://github.com/huggingface/diffusers/tree/v0.30.3)
- Requirements:
  - https://github.com/huggingface/diffusers/blob/v0.30.3/setup.py

### Explanation of changes:

- Skip py<38
- Update `huggingface_hub`'s pinning in `run`
- Add `safetensors >=0.3.1` to `run`
- Remove the `diffusers-cli --help` command from the `diffusers-base` subpackage because it already requires `pytorch` at runtime. I added a comment to `test/commands `of `diffusers-torch`.

### Notes:

-

[PKG-5909]: https://anaconda.atlassian.net/browse/PKG-5909?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml
Comment (scotthain) [inline recipe/meta.yaml:1]: Was the file permission change intentional?

File: abs.yaml
Comment (scotthain) [inline abs.yaml:1]: Is this file meant to be checked in?

**Review by scotthain — COMMENTED** [review]
minor questions, otherwise LGTM
🚀 🍰

**Review by scotthain — APPROVED** [review]
Again forgot to approve...
🚀 🍰

File: abs.yaml
Comment (skupr-anaconda) [reply abs.yaml:1]: If we build a chain of dependencies we can use `abs.yaml` with the `channels` key to pick up those dependencies from `https://staging.continuum.io/prefect/fs/ `+ `<PKG_NAME>-feedstock/pr<NUM>/<COMMIT>`

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml:1]: Hm... That's interesting. It wasn't intentional. I wonder what was the reason.

File: abs.yaml
Comment (cbouss) [reply abs.yaml:1]: This does need to be removed before merging though.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,13 +8,11 @@ package:
...
+  sha256: 67c5eb25d5b50bf0742624ef43fe0f6d1e1604f64aad3e8558469cbe89ecf72f
 
 build:
   number: 0
-  skip: true  # [py<37]
-  # pytorch is not built for python 3.11 on ppc64le
-  skip: true  # [py>310 and ppc64le]
+  skip: true  # [py<38]
 
 requirements:
```
Comment (cbouss) [inline recipe/meta.yaml]: The global requirements section should be removed, as it is not used (no build.sh or bld.bat).

File: recipe/meta.yaml [hunk 2]
```diff
@@ -32,9 +30,7 @@ outputs:
     build:
       entry_points:
         - diffusers-cli=diffusers.commands.diffusers_cli:main
-      skip: true  # [py<37]
-      # pytorch is not built for python 3.11 on ppc64le
-      skip: true  # [py>310 and ppc64le]
+      skip: true  # [py<38]
       missing_dso_whitelist:
```
Comment (cbouss) [inline recipe/meta.yaml]: Can you try removing all missing_dso_whitelist?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -50,32 +46,30 @@ outputs:
         - setuptools
         - wheel
         - pillow
-        - numpy
+        - numpy {{ numpy }}
```
Comment (cbouss) [inline recipe/meta.yaml]: I don't see pillow and numpy being needed as host dependencies. They can be removed.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -50,32 +46,30 @@ outputs:
...
         - python
         - filelock
-        - huggingface_hub >=0.13.2
+        - huggingface_hub >=0.23.2
         - importlib-metadata
         - numpy
         - pillow
         - regex !=2019.12.17
         - requests
+        - safetensors >=0.3.1
```
Comment (cbouss) [inline recipe/meta.yaml:40]: Can you add a run_constrained section following thse links: 
https://github.com/huggingface/diffusers/blob/v0.30.3/src/diffusers/dependency_versions_table.py/
https://github.com/huggingface/diffusers/blob/v0.30.3/setup.py#L211
https://github.com/huggingface/diffusers/blob/v0.30.3/setup.py#L231

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done

File: recipe/meta.yaml (hunk 4 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:40]: Done

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done

**Review by cbouss — APPROVED** [review]

**Pull request merged**